src/Common/src/System/Security/Cryptography/RSAOpenSsl.cs

```diff
@@ -130,6 +130,54 @@ public override byte[] Decrypt(byte[] data, RSAEncryptionPadding padding)
             SafeRsaHandle key = _key.Value;
             CheckInvalidKey(key);
 
+            int keySizeBytes = Interop.Crypto.RsaSize(key);
+
+            // OpenSSL does not take a length value for the destination, so it can write out of bounds.
+            // To prevent the OOB write, decrypt into a temporary buffer.
+            if (destination.Length < keySizeBytes)
+            {
+                Span<byte> tmp = stackalloc byte[0];
```

Comment:
why is the stackalloc needed here?

Reply:
without this the `tmp` variable is not allowed to receive the stackalloc from within the if (it's a compile error to remove this).